submissions/kt-std/friends-app/js/script.js

```diff
@@ -0,0 +1,455 @@
+    return ["list__label", "select__container", "number", "checkbox"]
+        .some((filter) => {
+            if (!(event.type === "click" &&  event.target.classList.contains("select__container"))) {
+                return event.target.classList.contains(filter);
+            }
+        });
+}
+
+function updateCards(event) {
+    if (checkFiltersChanged(event)) {
+        FRIENDS_ARRAY = INITIAL_FRIENDS_ARRAY;
+        sortCards(event, FRIENDS_ARRAY);
+        FRIENDS_ARRAY = filterByAge(FRIENDS_ARRAY);
+        FRIENDS_ARRAY = filterByGender(event, FRIENDS_ARRAY);
+        appendFriendsCards(FRIENDS_ARRAY);
+    }
+}
+
+function isSelectModified() {
+    return SELECT_CONTAINER.attributes["select-modified"].value === "true";
+}
+
+function checkOptionsVisibility() {
+    return OPTIONS_CONTAINER.classList.contains("visible");
+}
+
+function resetPreviouslySelectedOptions(selectedOption) {
+    Array.from(OPTIONS_LIST).forEach((option) => {
+        if (option.textContent != selectedOption) {
+            document.getElementById(option.htmlFor).checked = false;
+        }
+    });
+}
+
+function removeAriaSelectedAttribute(option) {
+    option.removeAttribute("aria-selected");
+}
+
+function getSelectOption(selectText) {
+    return Array.from(OPTIONS_LIST).find(
+        (option) => option.textContent === selectText
+    );
+}
+
+function higlightSelectedOption(itemToSelect) {
+    itemToSelect.setAttribute("aria-selected", "true");
+    document.getElementById(itemToSelect.htmlFor).checked = true;
+    updateSelectText(itemToSelect);
+}
+
+function focusOnItem(buttonPressed) {
+    const optionsListArray = Array.from(OPTIONS_LIST),
+        selectedItem = optionsListArray.find(
+            (listItem) => listItem.textContent === selectedOption
+        ),
+        selectedItemIndex = Array.from(OPTIONS_LIST).indexOf(selectedItem);
+    if (buttonPressed === "ArrowUp") {
+        switch (true) {
+            case !isFirstListItem(selectedItemIndex):
+                higlightSelectedOption(optionsListArray[selectedItemIndex - 1]);
+                selectedOption = optionsListArray[selectedItemIndex - 1].textContent;
+                resetPreviouslySelectedOptions(selectedOption);
+                removeAriaSelectedAttribute(selectedItem);
+                break;
+            case !isFirstListItem(selectedItemIndex):
+                break;
+        }
+    } else if (buttonPressed === "ArrowDown") {
+        switch (true) {
+            case !isLastListItem(selectedItemIndex, optionsListArray):
+                higlightSelectedOption(optionsListArray[selectedItemIndex + 1]);
+                selectedOption = optionsListArray[selectedItemIndex + 1].textContent;
+                resetPreviouslySelectedOptions(selectedOption);
+                removeAriaSelectedAttribute(selectedItem);
+                break;
+            case isLastListItem(selectedItemIndex, optionsListArray):
+                break;
+        }
```

Comment:
With `ArrowDown` and `ArrowUp` you can increment and decrement index.

Then check index for validity.

Then select item.